crates/flashblocks-rpc/src/cache.rs

```diff
@@ -0,0 +1,282 @@
+use alloy_consensus::Transaction as _;
+use alloy_consensus::transaction::SignerRecoverable;
+use alloy_consensus::transaction::TransactionMeta;
+use alloy_primitives::{Address, Sealable, TxHash, U256};
+use alloy_rpc_types::Withdrawals;
+use alloy_rpc_types::{BlockTransactions, Header, TransactionInfo};
+use op_alloy_consensus::OpTxEnvelope;
+use op_alloy_network::Optimism;
+use op_alloy_rpc_types::OpTransactionReceipt;
+use op_alloy_rpc_types::Transaction;
+use reth_optimism_chainspec::OpChainSpec;
+use reth_optimism_evm::extract_l1_info;
+use reth_optimism_primitives::{OpBlock, OpReceipt, OpTransactionSigned};
+use reth_optimism_rpc::OpReceiptBuilder;
+use reth_primitives::Recovered;
+use reth_primitives_traits::block::body::BlockBody;
+
+use reth_rpc_eth_api::{RpcBlock, RpcReceipt};
+use rollup_boost::{
+    FlashblockBuilder, FlashblocksPayloadV1, OpExecutionPayloadEnvelope, PayloadVersion,
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    str::FromStr,
+    sync::{Arc, Mutex},
+};
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct Metadata {
+    pub receipts: HashMap<String, OpReceipt>,
+    pub new_account_balances: HashMap<String, String>, // Address -> Balance (hex)
+    pub block_number: u64,
+}
+
+#[derive(Clone)]
+pub struct FlashblocksCache {
+    inner: Arc<Mutex<FlashblocksCacheInner>>,
```

Comment:
IMO thin is problematic design, because you lock whole structure when in fact you need only part of the structure.